make the REPL (`nim secret`) work again

zerbina opened the pull request:
## Summary

Properly configure the compiler when running the `nim secret`
command, making the compiler built-in REPL usable again.

## Details

* the compiler is configured for the VM backend when the
  `cmdInteractive` command is active
* `echo` output is no longer prefixed by `>>>`
* a test is added to ensure the basic REPL functionality works
* the standard output stream is flushed after writing `>>>` or `...`,
  so that the test is able to read the output

zerbina committed:
main: customize for the VM backend

For the `secret` command, customize for the VM backend, rather than for
the C backend (which leads to errors caused by unsupported code being
enabled).

zerbina committed:
cli_reporter: don't prefix echo output

The echo output getting prefixed with the input indicator ">>>" seems
to have been accidentally introduced in commit
f35b5bf2d447c10b6a104ef0649115a266e8dea6.

zerbina committed:
tests: add a test for `nim secret`

zerbina committed:
llstream: flush after writing

If the output stream is buffered, not flushing means that the test
wont receive the characters, and thus cannot work correctly.

saem commented on compiler/front/cli_reporter.nim:
```diff
@@ -2617,8 +2617,7 @@ To create a stacktrace, rerun compilation with './koch temp $1 <file>'
     of rintEchoMessage:
-      result = if conf.cmd == cmdInteractive: ">>> " & r.msg
-               else:                          r.msg
+      result = r.msg
```
Note to self remove reporting from this code path.

saem commented on tests/compilerfeatures/trepl.nim:
```diff
@@ -0,0 +1,54 @@
+discard """
+  description: "Ensure that the basic REPL functionality works"
+  targets: native
```
Should a timeout be added because this test is susceptible to bugs causing it to get stuck on reads?

zerbina replied:
Good call, I'll add a timeout. It currently won't protect against the test getting stuck, however, since testaement's `timeout` feature is only an "after the fact" check.

saem reviewed (approved):
Looks good, minor suggestion about adding a timeout to a test.

zerbina committed:
tests: add a timeout to `trepl.nim`

At the moment, this won't prevent the test getting stuck, but it will
once testament properly handles timeouts.

zerbina commented:
/merge
